lib/backend/migration/migration.go

```diff
@@ -0,0 +1,155 @@
+package migration
+
+import (
+	"context"
+	"sync/atomic"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/gravitational/teleport"
+	"github.com/gravitational/teleport/api/utils/retryutils"
+	"github.com/gravitational/teleport/lib/backend"
+	"github.com/gravitational/trace"
+)
+
+// Migration manages a migration between two [backend.Backend] interfaces.
+type Migration struct {
```

Comment:
What should we call it `Exporter`?

Reply:
@fspmarshall made a few suggestions below

 > nit: migrate is a pretty broad term for a top-level command. I'd lean toward something like either backend migrate or migrate backend so that we can introduce other backend and/or migration utilities without eating up our namespacing too much.

@rosstimothy What do you mean when you say `When most people think of migrations they associate it with migrating the teleport backend state`? Is this referring to changes to our data model or something else?

Reply:
👍 `teleport backend clone` sounds good to me.